connectors/src/connectors/zendesk/temporal/sync_ticket.ts

```diff
@@ -17,59 +36,164 @@ export async function syncTicket({
   currentSyncDateMs: number;
   loggerArgs: Record<string, string | number | null>;
   forceResync: boolean;
+  comments: ZendeskFetchedTicketComment[];
+  users: ZendeskFetchedUser[];
 }) {
   let ticketInDb = await ZendeskTicketResource.fetchByTicketId({
     connectorId,
     ticketId: ticket.id,
   });
+
   const createdAtDate = new Date(ticket.created_at);
   const updatedAtDate = new Date(ticket.updated_at);
 
+  const shouldPerformUpsertion =
+    forceResync ||
+    !ticketInDb ||
+    !ticketInDb.lastUpsertedTs ||
+    ticketInDb.lastUpsertedTs < updatedAtDate;
+
+  const commonTicketData = {
+    subject: ticket.subject,
+    url: ticket.url,
+    createdAt: createdAtDate,
+    updatedAt: updatedAtDate,
+    assigneeId: ticket.assignee_id,
+    groupId: ticket.group_id,
+    organizationId: ticket.organization_id,
+    lastUpsertedTs: new Date(currentSyncDateMs),
+  };
+
   if (!ticketInDb) {
     ticketInDb = await ZendeskTicketResource.makeNew({
       blob: {
-        createdAt: createdAtDate,
-        updatedAt: updatedAtDate,
+        ...commonTicketData,
         ticketId: ticket.id,
         brandId,
         permission: "read",
-        assigneeId: ticket.assignee_id,
-        groupId: ticket.group_id,
-        organizationId: ticket.organization_id,
-        name: "Ticket",
-        description: ticket.description,
-        subject: ticket.subject,
-        requesterMail: ticket?.requester?.email || "",
-        url: ticket.url,
-        lastUpsertedTs: new Date(currentSyncDateMs),
-        satisfactionScore: ticket.satisfaction_rating?.score || "unknown",
-        satisfactionComment: ticket.satisfaction_rating?.comment || "",
-        status: ticket.status,
-        tags: ticket.tags,
-        type: ticket.type,
-        customFields: ticket.custom_fields.map((field) => field.value),
         connectorId,
       },
     });
   } else {
-    await ticketInDb.update({
+    await ticketInDb.update(commonTicketData);
+  }
+
+  if (!shouldPerformUpsertion) {
+    logger.info(
+      {
+        ...loggerArgs,
+        connectorId,
+        ticketId: ticket.id,
+        ticketUpdatedAt: updatedAtDate,
+        dataSourceLastUpsertedAt: ticketInDb?.lastUpsertedTs ?? null,
+      },
+      "[Zendesk] Ticket already up to date. Skipping sync."
+    );
+    return;
+  } else {
+    logger.info(
+      {
+        ...loggerArgs,
+        connectorId,
+        ticketId: ticket.id,
+        ticketUpdatedAt: updatedAtDate,
+        dataSourceLastUpsertedAt: ticketInDb?.lastUpsertedTs ?? null,
+      },
+      "[Zendesk] Ticket to sync."
+    );
+  }
+
+  if (ticket.description || comments.length > 0) {
+    const metadata = [
+      `Ticket ID: ${ticket.id}`,
+      `Subject: ${ticket.subject}`,
+      `Created At: ${createdAtDate.toISOString()}`,
+      `Updated At: ${updatedAtDate.toISOString()}`,
+      `Status: ${ticket.status}`,
+      `Priority: ${ticket.priority || "N/A"}`,
```

Comment:
open question: should we add an N/A priority and type or filter them out like the satisfaction rating?